src/services/sessions.js

```diff
@@ -1225,9 +1236,14 @@ module.exports = class SessionsHelper {
 			// If enrolled by the mentee get email and name from user service via api call.
 			// Else it will be available in userTokenData
 			if (isSelfEnrolled) {
-				const userDetails = (await userRequests.fetchUserDetails({ userId: userTokenData.id })).data.result
-				userId = userDetails.id
-				email = userDetails.email
+				const userDetails = await mentorExtensionQueries.getMentorExtension(
+					userTokenData.id,
+					['user_id', 'name', 'email'],
+					true
+				)
+
+				userId = userDetails.user_id
+				email = await emailEncryption.decrypt(userDetails.email)
```

Comment:
@rakeshSgr we can put this in a common place rather than doing it everywhere maybe in pushEmailToKafka function?